src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -504,13 +511,18 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 		this._model.updateActiveCell(this.cellModel);
 	}
 
-	private findPathRelativeToContent(elementContent: string): string {
+	private findPathRelativeToContent(absolutePathURI: URI): string {
 		let notebookFolder = this.notebookUri ? path.join(path.dirname(this.notebookUri.fsPath), path.sep) : '';
 		if (notebookFolder) {
-			let absolutePathURI = URI.parse(elementContent);
 			if (absolutePathURI?.scheme === 'file') {
 				let relativePath = path.relative(notebookFolder, absolutePathURI.fsPath);
-				return relativePath ? relativePath : '';
+				//if path contains whitespaces then it's not identified as a link
+				relativePath = relativePath.replace(/\s/g, '%20');
+				if (relativePath.startsWith(path.join('.', path.sep))) {
```

Comment:
path.relative can start with `../` as well - are you purposely not checking for that?